rbac/management/role/definer.py

```diff
@@ -274,3 +282,102 @@ def delete_permission(permission: Permission):
             dual_write_handler.replicate_update_system_role()
         else:
             dual_write_handler.replicate_new_or_updated_role(role)
+
+
+def _seed_v2_role_from_v1(v1_role, display_name, description, public_tenant, platform_roles):
+    """Create or update V2 role from V1 role during seeding."""
+    try:
+        v2_role, v2_created = SeededRoleV2.objects.update_or_create(
+            uuid=v1_role.uuid,
+            defaults={
+                "name": display_name,
+                "description": description,
+                "tenant": public_tenant,
+                "v1_source": v1_role,
+            },
+        )
+
+        if v2_created:
+            logger.info("Created V2 system role %s.", display_name)
+        else:
+            logger.info("Updated V2 system role %s.", display_name)
+
+        v2_role.permissions.clear()
+        v1_permissions = [access.permission for access in v1_role.access.all()]
+        if v1_permissions:
+            v2_role.permissions.set(v1_permissions)
+            logger.info("Added %d permissions to V2 role %s.", len(v1_permissions), display_name)
+
+        resource_service = ImplicitResourceService.from_settings()
+        scope = resource_service.scope_for_role(v1_role)
+
+        if v1_role.platform_default:
+            platform_role = platform_roles.get((DefaultAccessType.USER, scope))
+            if platform_role:
+                platform_role.children.add(v2_role)
+                logger.info("Added %s as child of platform role %s", display_name, platform_role.name)
+            else:
+                logger.warning(
+                    "Platform role for %s %s scope not available. Skipping child relationship for %s",
+                    DefaultAccessType.USER.value,
+                    scope.value,
+                    display_name,
+                )
```

Comment:
We also need to remove any existing parent–child relationship if the role is not `platform_default` (and similarly for `admin_default` below).